### PR TITLE
Database testing

### PR DESCRIPTION
**Pull request by BenNewman90**
- I've redone the database to use pg-promise rather than pg. 
- I've done the database testing (using promises).

### CODE REVIEW

File: server/database/build.js [hunk 1]
```diff
@@ -1,82 +1,65 @@
...
+      user.photo,
+      user.groups,
+      user.created,
+      user.last_login,
+      user.predictions,
+      user.points,
+      user.paid,
+    ])),
+  );
+  console.log(users);
```
Comment (Karyum) [inline server/database/build.js]: do you need that console log here?

File: server/database/dbconnection.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+const pgp = require('pg-promise')();
+require('env2')('./config.env');
 
-let { DEV_DB_URL } = process.env;
-if (process.env.NODE_ENV === "test") {
-  DB_URL = process.env.TEST_DB_URL;
-}
+let { DB_URL } = process.env;
+const { NODE_ENV } = process.env;
+const { TEST_DB_URL } = process.env;
```
Comment (Karyum) [inline server/database/dbconnection.js]: oyy you could just do `{DB_URL, NODE_ENV, TEST_DB_URL } = process.env` and still would work the same

File: testing/database_tests.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const { db } = require('../server/database/dbconnection');
+
+tape('--------------database_tests.js----------check tape is working', (t) => {
+  t.ok(true, 'tape is working');
+  t.equal(process.env.NODE_ENV, 'test', 'The process.env.NODE_ENV environment should be test');
+  t.end();
+});
+
+tape('testing runDbBuild', (t) => {
+  buildDatabase()
```
Comment (Karyum) [inline testing/database_tests.js]: it's much better for you and for you testing environment sanity to build the db once and then just test, i would say consult with @shiryz on that

File: server/database/dbconnection.js (hunk 2 above)
Comment (BenNewman90) [reply server/database/dbconnection.js]: Good point. updated.

File: testing/database_tests.js (hunk 3 above)
Comment (BenNewman90) [reply testing/database_tests.js]: I've had a look at this but will leave it for now.. it will be in the next pull request.

Comment (Karyum) [reply testing/database_tests.js]: did you ask @shiryz

Comment (BenNewman90) [reply testing/database_tests.js]: Yes. I've spoken to her. I understand how to do it.. will do it this eve.

**Pull request merged**